plugins/destination/mongodb/client/write.go

```diff
@@ -2,63 +2,144 @@ package client
 
 import (
 	"context"
+	"encoding/json"
 
-	"github.com/cloudquery/plugin-sdk/schema"
-	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/cloudquery/plugin-sdk/v2/schema"
+	"github.com/cloudquery/plugin-sdk/v2/specs"
+	"github.com/cloudquery/plugin-sdk/v2/types"
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/mongo"
 )
 
-func (c *Client) appendTableBatch(ctx context.Context, table *schema.Table, resources [][]any) error {
-	documents := make([]any, len(resources))
-	for i, resource := range resources {
-		document := make(bson.M, len(table.Columns))
-		for l, col := range table.Columns {
-			document[col.Name] = resource[l]
+func transformArr(arr arrow.Array) []any {
+	dbArr := make([]any, arr.Len())
+	for i := 0; i < arr.Len(); i++ {
+		if arr.IsNull(i) || !arr.IsValid(i) {
+			dbArr[i] = nil
+			continue
 		}
-		documents[i] = document
+		switch a := arr.(type) {
+		case *array.Boolean:
+			dbArr[i] = a.Value(i)
+		case *array.Int16:
+			dbArr[i] = a.Value(i)
+		case *array.Int32:
+			dbArr[i] = a.Value(i)
+		case *array.Int64:
+			dbArr[i] = a.Value(i)
+		case *array.Float32:
+			dbArr[i] = a.Value(i)
+		case *array.Float64:
+			dbArr[i] = a.Value(i)
+		case *array.Binary:
+			dbArr[i] = a.Value(i)
+		case *array.LargeBinary:
+			dbArr[i] = a.Value(i)
+		case *array.String:
+			dbArr[i] = a.Value(i)
+		case *array.LargeString:
+			dbArr[i] = a.Value(i)
+		case *array.Timestamp:
+			dbArr[i] = a.Value(i).ToTime(arrow.Microsecond)
+		case *types.JSONArray:
+			var val any
+			if err := json.Unmarshal([]byte(a.ValueStr(i)), &val); err != nil {
+				panic(err)
+			}
+			dbArr[i] = val
+		case array.ListLike:
+			start, end := a.ValueOffsets(i)
+			nested := array.NewSlice(a.ListValues(), start, end)
+			dbArr[i] = transformArr(nested)
+			nested.Release()
+		default:
+			dbArr[i] = arr.ValueStr(i)
```

Comment:
Will this handle structs in the way we want? Maybe we should use json marshaling here, rather than value string?

Reply:
structs are handled under `JSON` . Otherwise we will have the same problem for example with UUID having`"` if we wont use `ValueStr` or any other unknown type might have un-necessary `"`

Reply:
Ah, what I mean is what happens if we receive an arrow struct type: even though source plugins don't do that today, that case is not handled explicitly here so we would have to come back and update it later it seems.